### DISCUSSION TITLE
Graph X scale/position are adversely affected by longer-acting insulins

### DISCUSSION THREAD

**Discussion by thedewi** [NightscoutFoundation/xDrip#2866] [General]
I like to use Predictive Simulations with a longer acting insulin profile (FIASP), but this adversely affects the graph's default viewport in a few ways. The graph tries to show the long tail of insulin action, which means that:

1. The X scale is "bunched up", showing 5+ hours, giving the impression that my level is changing more rapidly than it is.
2. My locked time button is ignored.
3. When I manually press a time button, the X position emphasizes the tail of insulin action, so I have to scroll left to see my blood data.
4. The default viewport is very aggressive in re-asserting itself: it happens whenever the chart updates, and when we switch apps. I open xDrip+ 50 times every day so the manual adjustments really add up.

**Comment by thedewi** [comment]
5. Predicted lows also sometimes affect the X scale and position, and I'd prefer they didn't, as the jarring changes in scale and position makes them hard to interpret.

### What I propose

1. Time button locks should ideally apply even when showing insulin action.
2. The default X position should ideally be oriented around the most recent readings, not the time scale of future predictions.
3. I think a good position would place the most recent BGL reading 1/3 from the right edge of the graph, so we can see 2/3 history and 1/3 prediction.

**Reply by Navid200** [reply]
I'm OK with your 1.
It has been my plan to open a PR to propose that.

**Comment by thedewi** [comment]
### Current default view: 8 hour scale despite time button lock

![Screenshot_20230602-102443](https://github.com/NightscoutFoundation/xDrip/assets/4726656/5b2b3469-01c7-4653-ad3c-76e8455478e8)

### After tapping the time button: data not visible

![Screenshot_20230602-102459](https://github.com/NightscoutFoundation/xDrip/assets/4726656/c5a63ba6-95d3-473e-b4e9-ccd957864ea7)
